src/backend/interfaces/REST/go.server/middlewares/authentication.go

```diff
@@ -5,16 +5,61 @@
 package http_middleware
 
 import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/sha256"
+	"encoding/asn1"
 	"encoding/base64"
+	"errors"
+	"log"
+	"math/big"
 	"net/http"
 	"strconv"
 	"strings"
 	"time"
-
 	"github.com/CaliOpen/Caliopen/src/backend/main/go.backends"
 	"github.com/gin-gonic/gin"
 )
 
+type ecdsaSignature struct {
+	R, S *big.Int
+}
+
+// getSignedQuery, build the HTTP query that has been signed
+func getSignedQuery(c * gin.Context) string {
+	query := c.Request.Method + c.Request.URL.String()
+	return query
+}
+
+// verifySignature, check for validity of device ecdsa signature
+func verifySignature(signature string, query string, curve string, x big.Int, y big.Int) (bool, error) {
```

Comment:
shorter func signature could be : `verifySignature(signature, query, curve string, x, y big.Int) (bool, error) {`

Reply:
I wasn't aware of this shortcut. Thanks I rewrite using this syntax